tests/drivers/adc/adc_error_cases/src/adc_error_cases.c

```diff
@@ -0,0 +1,244 @@
+		" got unexpected value of %d",
+		ret
+	);
+}
+
+/**
+ * @brief test adc_read() with invalid channels value
+ *
+ * function should return -EINVAL
+ */
+
+ZTEST(adc_error_cases, test_adc_read_invalid_channels)
+{
+	int ret;
+
+	adc_channel_setup(dev_adc, &valid_channel_cfg);
+
+	struct adc_sequence invalid_seq = valid_seq;
+	/* Set channels configuration to invalid value */
+	invalid_seq.channels = 0;
+
+	ret = adc_read(dev_adc, &invalid_seq);
+
+	zassert_true(
+		ret == -EINVAL,
+		"adc_read() should return -EINVAL,"
+		" got unexpected value of %d",
+		ret
+	);
+}
+
+/**
+ * @brief test adc_read() with not configured channel
+ *
+ * function should return -EINVAL
+ */
+
+ZTEST(adc_error_cases, test_adc_read_not_configured_channel)
+{
+	int ret;
+
+	adc_channel_setup(dev_adc, &valid_channel_cfg);
+
+	struct adc_sequence invalid_seq = valid_seq;
+	/* Set channels configuration to use not configured channel */
+	invalid_seq.channels = 2;
+
+	ret = adc_read(dev_adc, &invalid_seq);
+
+	zassert_true(
+		ret == -EINVAL,
+		"adc_read() should return -EINVAL,"
+		" got unexpected value of %d",
+		ret
+	);
+}
+
+/**
+ * @brief test adc_read() with invalid buffer length
+ *
+ * function should return -ENOMEM
+ */
+
+ZTEST(adc_error_cases, test_adc_read_invalid_buffer)
+{
+	int ret;
+
+	adc_channel_setup(dev_adc, &valid_channel_cfg);
+
+	struct adc_sequence invalid_seq = valid_seq;
+	/* set buffer size to 0 bytes */
+	invalid_seq.buffer_size = 0;
+
+	ret = adc_read(dev_adc, &invalid_seq);
+
+	zassert_true(
+		ret == -ENOMEM,
+		"adc_read() should return -ENOMEM,"
+		" got unexpected value of %d",
+		ret
+	);
+}
+
+/**
+ * @brief test adc_channel_setup() with invalid reference value
+ *
+ * function should return -EINVAL
+ */
+
+ZTEST(adc_error_cases, test_adc_setup_invalid_reference)
+{
+	int ret;
+
+	struct adc_channel_cfg invalid_channel_cfg = valid_channel_cfg;
+	/* set invalid reference */
+	invalid_channel_cfg.reference = 99;
+
+	ret = adc_channel_setup(dev_adc, &invalid_channel_cfg);
+
+	zassert_true(
+		ret == -EINVAL,
+		"adc_channel_setup() should return -EINVAL,"
+		" got unexpected value of %d",
+		ret
+	);
+}
+
+/**
+ * @brief test adc_read() with invalid gain value
+ *
+ * function should return -EINVAL
+ */
+
+ZTEST(adc_error_cases, test_adc_setup_invalid_gain)
+{
+	int ret;
+
+	struct adc_channel_cfg invalid_channel_cfg = valid_channel_cfg;
+	/* set invalid gain value */
+	invalid_channel_cfg.gain = 99;
+	ret = adc_channel_setup(dev_adc, &invalid_channel_cfg);
+	zassert_true(
+		ret == -EINVAL,
+		"adc_channel_setup() should return -EINVAL,"
+		" got unexpected value of %d",
+		ret
+	);
+}
+
+/**
+ * @brief test adc_read() with invalid acquisition time
+ *
+ * function should return -EINVAL
+ */
+
+ZTEST(adc_error_cases, test_adc_setup_invalid_acquisition_time)
+{
+	int ret;
+
+	struct adc_channel_cfg invalid_channel_cfg = valid_channel_cfg;
+	/* set invalid acquisition_time value */
+	invalid_channel_cfg.acquisition_time = 16384;
```

Comment:
Why should that be invalid acquisition time?

Reply:
Honestly - don't remember, changed to `ADC_ACQ_TIME_MAX + 1`

Reply:
It's still a valid acquisition time specification (meaning 0 microseconds), though the Nordic driver returns `-EINVAL` for it. 
I think this test case should just be removed.

Reply:
ok, removed